packages/truffle/test/sources/migrations/typescript/migrations/2_migrations_sync.ts

```diff
@@ -0,0 +1,10 @@
+/// <reference types="./global" />
+/* global artifacts */
```

Comment:
The `reference types` line here won't be necessary in the final PR. I'm also going to try to find a way around the `global artifacts` comment so users who are using ESLint don't need to add that, but I have lower confidence on that one.

Reply:
We have `.eslint` [rules for this that target `*.js` files](https://github.com/trufflesuite/truffle/blob/develop/.eslintrc.truffle.json#L8-L16), perhaps extending the glob to include typescript would suffice?